backend/telemetry/src/lib.rs

```diff
@@ -19,6 +20,20 @@ mod built_info {
     include!(concat!(env!("OUT_DIR"), "/built.rs"));
 }
 
+#[derive(Debug, Error)]
+pub enum TelemetryError {
+    #[error("Could not create MetricExporter: {0}")]
+    MetricExporterBuildError(opentelemetry_sdk::metrics::MetricError),
+    #[error("Span Exporter Build error: {0}")]
+    SpanExporterBuildError(#[from] opentelemetry::trace::TraceError),
+    #[error("Tracing Subscriber initialization error: {0}")]
+    TracingSubscriberInitError(#[from] tracing_subscriber::util::TryInitError),
+    #[error("Tracer shutdown error: {0}")]
+    TracerShutdownError(opentelemetry::trace::TraceError),
+    #[error("Meter shutdown error: {0}")]
+    MeterShutdownError(#[from] opentelemetry_sdk::metrics::MetricError),
+}
```

Comment:
It's probably worth de-duplicating the common error types, they should contain enough information within to tell if it's from build or shutdown